React/nested-event-handling.md

```diff
@@ -14,156 +14,173 @@ By the end of this lesson, students should be able to...
 </details>
 
+We can now remove `useState` from the `Student` component and return all references to the state variables into props.
+
 Ultimately, we can refactor the code to look like this:
 
 ```javascript
-// Student.js
+// src/components/Student.js
+// ...
 const Student = (props) => {
+
+  // Event callback functions
+  const onButtonClick = () => {
+    // Updated below
+  }
+
+  const onFullNameInputChange = (event) => {
+    // Updated below
+  };
+
+  // Component functions always return JSX
   return (
-    <section className= 'student'>
-      <h3>Student Component</h3>
-      <h4 >Name {props.fullName} </h4>
-      <p>Email: {props.email} </p>
-    </section>
+    <div>
+      <h3>{props.fullName}</h3>
+      <input value={props.fullName} onChange={onFullNameInputChange} />
+      <ul>
+        <li>Class: C13</li>
+        <li>Birthday: {props.birthday}</li>
+        <li>Email: {props.email}</li>
+      </ul>
+      <button onClick={onButtonClick}>
+        Mark {props.present ? 'Absent' : 'Present'}
+      </button>
+    </div>
   );
-}
+};
 ```
 
-## Modify `state` using an event
-Listing out our students is great, but what if we could also track attendance? Let's do it! We'll set up a button for each student. When pressed, this button will update the `state` data for that student to mark them as "present". Additionally, we'll add some CSS so that students who have been marked "present" will be identified to the user.
+## Prop callback functions
+
+So now our Student component gets information from `App` and `StudentCollection`, but still needs to tell `App` when to change this particular student.  Additionally, we'll add some CSS so that students who have been marked "present" will be identified to the user.
 
 There are a few things to consider when making this change to our application. Think about these questions with your neighbors.
+
 1. Which component should contain the button that will enable us to mark an individual student as "present"?
 2. Which component is managing the student data?
 3. Is the answer to #1 the same as #2?
 
-If you answered NO to question #3, you're on to something big. The data related to the students is tracked in the `StudentCollection` while the button to mark an individual student "present" should really be on each individual `Student` component. Our challenge then is to use the tools we have been given to _propagate_ the button press event from one component to another.
+If you answered NO to question #3, you're on to something big. The data related to the students is tracked in `App` while the button to mark an individual student "present" should really be on each individual `Student` component. Our challenge then is to use the tools we have been given to _propagate_ the button press event from one component to another.
```

Comment:
I'm inclined to suggest hiding this with a details/summary block since it feels a lot like an answer to a question we want them to contemplate on their own

Reply:
I think you missed this since github decided to collapse it.